model/textparse/openmetricsparse.go

```diff
@@ -222,9 +225,108 @@ func (p *OpenMetricsParser) Exemplar(e *exemplar.Exemplar) bool {
 // CreatedTimestamp returns nil as it's not implemented yet.
 // TODO(bwplotka): https://github.com/prometheus/prometheus/issues/12980
 func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
+	var lbs labels.Labels
+	p.Metric(&lbs)
+	lbs = lbs.DropMetricName()
+
+	newParser := deepCopyParser(p)
+loop:
+	for {
+		switch t, _ := newParser.Next(); t {
+		case EntrySeries:
+			// TODO: potentially broken? Missing type?
+			if newParser.mName != p.mName {
+				return nil
+			}
+
+			var newLbs labels.Labels
+			newParser.Metric(&newLbs)
+			name := newLbs.Get(model.MetricNameLabel)
+			if !strings.HasSuffix(name, "_created") {
+				continue
+			}
+
+			// edge case: if gauge_created of unknown type -> skip parsing
+			newLbs = newLbs.DropMetricName()
```

Comment:
I've noticed there's no method to drop labels by labelName, we would need to implement it... but after that, the whole switch case could be replaced with:

```go
// Reserved labels that _created lines don't have when compared to original metric.
newLbs = newLbs.Drop("le", "quantile", labels.MetricName)
if !labels.Equal(lbs, newLbs) {
	// New Parser found new time series, meaning we don't have CT.
	return nil
}
```

Reply:
might still need to keep the switch (since it also checks the `mtype`) but it can still be simplified:
```go
// Reserved labels that _created lines don't have when compared to original metric.
newLbs = newLbs.Drop("le", "quantile", labels.MetricName)
switch p.mtype {
case model.MetricTypeCounter, model.MetricTypeSummary, model.MetricTypeHistogram:
    if !labels.Equal(lbs, newLbs) {
        // New Parser found new time series, meaning we don't have CT.
        return nil
    }
default:
    return nil
```

Reply:
or we could just move the switch into a separate conditional

Reply:
Didn't quite need to make a new function called drop. ended up doing this:
```go
switch p.mtype {
    case model.MetricTypeCounter, model.MetricTypeHistogram, model.MetricTypeSummary:
        labelDiffs := lbs.MatchLabels(false, newLbs.ExtractNames()...)
        if labelDiffs.Len() != 0 {
            if !labelDiffs.Contains("quantile", "le") || labelDiffs.Len() != 1 {
                return nil
             }
         }
    default:
    // if mtype is not counter, summary or histogram, it won't have a valid _created line
        return nil
}
```